client/scss/components/_dropdown.scss

```diff
@@ -99,3 +99,35 @@
     opacity: 1;
   }
 }
+
+.w-dropdown__toggle {
+  @apply w-p-2 w-text-text-label w-bg-transparent;
+}
+
+.w-dropdown__toggle--icon {
+  @apply w-w-8 w-h-8;
+}
+
+.w-dropdown__toggle-icon {
+  @apply w-w-4 w-h-4 w-align-text-bottom w-transition motion-reduce:w-transition-none;
+}
+
+.w-dropdown__content {
+  @apply w-flex w-flex-col w-justify-start w-py-2;
+}
+
+.w-dropdown__content :where(a, button) {
+  @apply w-bg-transparent w-inline-flex w-items-center w-font-bold w-text-text-label-menus-default w-py-3 w-px-6 w-no-underline w-transition w-outline-offset-inside;
+
+  .icon {
+    @apply w-w-4 w-h-4 w-mr-3 w-transition w-opacity-50;
+  }
```

Comment:
The button needs `text-align: left;`, and the icon needs `flex-shrink: 0;`.

```suggestion
.w-dropdown__content :where(a, button) {
  @apply w-bg-transparent w-inline-flex w-items-center w-text-left w-font-bold w-text-text-label-menus-default w-py-3 w-px-6 w-no-underline w-transition w-outline-offset-inside;

  .icon {
    @apply w-w-4 w-h-4 w-mr-3 w-transition w-opacity-50 w-shrink-0;
  }
```

### Before

<img width="527" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/c295dfec-03e2-415c-b6f0-094f0058171b">


### After

<img width="527" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/3353efba-c1b0-4713-83e7-4b62b8f98ec0">

Reply:
Good spotting, thank you.